pkg/dockerregistry/server/storagedriver.go

```diff
@@ -0,0 +1,32 @@
+package server
+
+import (
+	"regexp"
+
+	context "github.com/docker/distribution/context"
+	storagedriver "github.com/docker/distribution/registry/storage/driver"
+	registrystorage "github.com/docker/distribution/registry/storage/driver/middleware"
+)
+
+func init() {
+	registrystorage.Register("openshift", registrystorage.InitFunc(newLocalStorageDriver))
+}
+
+type localStorageDriver struct {
+	storagedriver.StorageDriver
+}
+
+var _ storagedriver.StorageDriver = &localStorageDriver{}
+
+var pathSplitRe = regexp.MustCompile("^(?P<repoPrefix>.*/repositories)/(?P<name>[^/]+/[^/]+)/(?P<content>.*)$")
```

Comment:
> pretty sure we don't want to use a regex here...

We can't avoid regex because we don't have access to storage configuration and do not know `rootDirectory`.

Basically path looks like this: `<root>/v2/repositories/<name>/_manifests/revisions/<algorithm>/<hex digest>/link` . We can remove `<root>` by regexp and split other part by slashes.

But in this case we will hardcode the directory structure in isolation from the `pathFor`. If the structure is changed, then our solution will fail silently. My solution changes only part of path almost without assuming its structure.